daemon/container_operations.go

```diff
@@ -455,6 +482,28 @@ func (daemon *Daemon) allocateNetwork(ctx context.Context, cfg *config.Config, c
 		}
 	}()
 
+	if err := ctr.BuildHostnameFile(); err != nil {
+		return nil, err
+	}
+
+	// TODO(robmry) - on Windows, running this after the task has been created does something
+	//  strange to the resolver ... addresses are assigned properly (including addresses
+	//  specified in the 'run' command), nslookup works, but 'ping' doesn't find the address
+	//  of a container. There's no query to our internal resolver from 'ping' (there is from
+	//  nslookup), so Windows must have squirreled away the address somewhere else.
+	if runtime.GOOS == "windows" {
+		if err := daemon.allocateNetwork(ctx, cfg, ctr); err != nil {
+			return nil, err
+		}
+	}
+
+	return newSandbox, nil
+}
+
+func (daemon *Daemon) allocateNetwork(ctx context.Context, cfg *config.Config, ctr *container.Container) (retErr error) {
+	start := time.Now()
+
+	// An intermediate map is necessary because "connectToNetwork" modifies "container.NetworkSettings.Networks"
```

Comment:
Oh, that comment was re-added in the 4th commit. You can forget about my previous comment.